exercises/dolica_akelloegwel/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,173 @@
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+#include <cctype>
+#include <algorithm>
+#include <vector>
+#include <set>
+
+bool ContainsDashes(std::string compositeWord)
+{
+	return compositeWord.find("-") != std::string::npos;
+}
+std::vector<std::string> SplitWords(std::string compositeWord)
+{
+	// Containers for splitting up words
+	std::vector<std::string> words;
+	std::string segment;
+
+	// Convert the composite word to a stream
+	std::stringstream wordStream(compositeWord);
+
+	// Split the elements of the stream using the '-' character and place these in the vector
+	while (std::getline(wordStream, segment, '-'))
+	{
+		// Ignore empty words
+		if (segment.length() == 0)
+			continue;
+
+		// Add the word to the vector
+		words.push_back(segment);
+	}
+
+	return words;
+}
+std::string PrepareWord(std::string rawWord)
+{
+	// Declare a string of punctuation characters
+	std::string unwantedChars = ".,?'\"!():;";
+
+	// Remove the unwanted characters from the string
+	for (int i = 0; i < unwantedChars.length(); i++)
+	{
+		rawWord.erase(std::remove(rawWord.begin(), rawWord.end(), unwantedChars[i]), rawWord.end());
+	}
+
+	// Convert the word to lower case
+	for (unsigned int i = 0; i < rawWord.length(); ++i)
+	{
+		rawWord[i] = std::tolower(rawWord[i]);
+	}
+
+	return rawWord;
+}
+void SaveWordCountToFile(std::map<std::string, int> wordMap, std::string outputFilename)
+{
+	// Attempt to open a file
+	std::ofstream outFile;
+	outFile.open(outputFilename);
+
+	// Check that the file was opened successfully
+	if (!outFile)
+	{
+		std::cout << "Error: Unable to open the file " << outputFilename << " for writing." << std::endl;
+		exit(1);
+	}
+
+	// Declare a comparator for sorting the map by word count
+	auto cmp = [](const auto &p1, const auto &p2)

```

Comment:
I would extract this as a named function at the point I was assigning the lambda to a variable. Things start to get messy otherwise.